Added option --debug_hosts to run debug on hosts

amar266 opened the pull request:
Added --debug_hosts option , which will enable debug_timeout function to print logs
Right Now It will only print few  lines from end of /var/log/syslog

bodepd commented:
can you mention what the motivation for this is?

bodepd commented:
I think something to pull out log files is better as a separate function. You should also make the name of the file configurable, b/c we might want to get /var/log/syslog or /var/log/cloud-ini.log or any arbitrary logs

amar266 commented:
i changed in deploy.sh so that,  through ssh_config ,a command  can be executed on all hosts (like "python -m jiocloud.orchestrate debug_timeout jenkins-puppet-rjil-gate-694 --debug_hosts")on all hosts, so for that   i changed the function and added an arg that  will only print syslog information for time being ,and skip the other logs like failures and registered host.
 
I was also thinking of any arbitrary logs, but  for that we need to change deploy.sh which will  accept an argument for which files we need to see in console.And the arg will be passed to the function.
I need suggetions.

bodepd commented:
I'm not sure about that requiring an option to be passed in. Each role might have different logs. Maybe this is something that needs to be configured by puppet.

amar266 commented:
I think use of puppet is a good idea.I will check how we can implement this.

amar266 commented:
@bodepd   For arbitrary logs , i am thinking of creating a class like
class rjil::debug_timeout(
  $debug_files ,
) {
  ini_setting { 'debug_timeout':
    path    => '/tmp/foo.ini',
    section => main,
    setting => files,
    value   => debug_files
  } 
}   
Which will create a ini file to specific location on all hosts and there would entry in hiera/data/common.yaml and role based override in roles folder like in hiera/data/role/gcp.yaml
rjil::debug_timeout::debug_files: "/var/log/syslog:80 /var/log/auth.log:80"

So the above will create a ini file , which can be parsed in debug_timeout function,

The Format /var/log/syslog:80 is "file_name:last n chars"

What do you think?

amar266 commented:
*value => debug_files  is value => $debug_files
